File: builder/java/Director.java
package builder.java;

public class Director {
    private AppleConsumer appleconsumer;

    void set_consumer(AppleConsumer consumer) {
        this.appleconsumer = consumer;
    }

    void consume_fruit() {
        appleconsumer.Peeler();
        appleconsumer.Knife();
    }
}

File: builder/rust/builder/src/director/director.rs
use crate::consumer::consumer::{Consumer};

pub struct Director {
}

impl Director {
    pub fn start_consuming(&self, consumer: &dyn Consumer ) {
        println!("{}", consumer.peeler().to_string());
        println!("{}", consumer.knife().to_string());
        println!("{}", consumer.juicer().to_string());

    }
}